Content.Shared/Actions/SharedActionsSystem.cs

```diff
@@ -596,6 +596,66 @@ public void PerformAction(Entity<ActionsComponent?> performer, Entity<ActionComp
         var performed = new ActionPerformedEvent(performer);
         RaiseLocalEvent(action, ref performed);
     }
+
+    /// <summary>
+    /// Attempts to perform an action, bypassing validation checks. Returns false on failure, true on success.
+    /// </summary>
+    /// <param name="performer">The entity performing the action</param>
+    /// <param name="action">The action being performed</param>
+    /// <param name="actionEvent">An event override to perform. If null, uses <see cref="GetEvent"/></param>
+    /// <param name="predicted">If false, prevents playing the action's sound on the client</param>
+    public bool TryPerformAction(Entity<ActionsComponent?> performer, Entity<ActionComponent> action, BaseActionEvent? actionEvent = null, bool predicted = true)
```

Comment:
Can't you use TryPerformAction on line 278? Forgot if we talked about this. Feels weird duplicating all this code here when that exists.

Reply:
Could you go into more detail? What didn't work about it and why did yours work?